docs/apache-airflow/concepts/taskflow.rst

```diff
@@ -81,6 +81,49 @@ To use logging from your task functions, simply import and use Python's logging
 
 Every logging line created this way will be recorded in the task log.
 
+Passing Arbitrary Objects As Parameters
+---------------------------------------
+
+.. versionadded:: 2.5.0
+
+As mentioned TaskFlow uses XCom to pass variables to each task. This requires that variables that are used as parameters
+to tasks need to be able to be serialized. Airflow out of the box supports all built-in types (like int or str) and it
+supports objects that are decorated with ``@dataclass`` or ``@attr.define``, but it will need some help if you want to
+pass custom objects or if you want to control serialization yourself. To do so add the ``serialize()`` method to your
+class and the staticmethod ``deserialize(data: dict, version: int)`` to your class. Like so:
+
+.. code-block:: python
+
+    from typing import ClassVar
+
+
+    class MyCustom:
+        version: ClassVar[int] = 1
+
+        def __init__(self, x):
+            self.x = x
+
+        def serialize(self) -> dict:
+            return dict({"x": self.x})
+
+        @staticmethod
+        def deserialize(data: dict, version: int):
+            if version > 1:
+                raise TypeError(f"version > {MyCustom.version}")
+            return MyCustom(data["x"])
```

Comment:
Could you demonstrate a taskflow task using the `MyCustom` class in the example?

Reply:
Are you open to "Tabbed" examples yet? :-)

Reply:
Referring to https://github.com/apache/airflow/pull/26486 :-) ?

Reply:
Yes, but I am not ready for that yet :-). Example added.

Reply:
I meant an example using this custom class. I'm not reading anything in relation to datasets. As a user, what do I do with `MyCustom`?